pallets/system/src/submitted_data.rs

```diff
@@ -107,17 +109,26 @@ fn root<I: Iterator<Item = Vec<u8>>>(submitted_data: I, metrics: RcMetrics) -> H
 	root
 }
 
-/// Calculates the merkle root using `Sha256` and `rs_merkle` crate.
+/// Calculates the merkle root using `Keccak256` and `rs_merkle` crate.
 #[cfg(feature = "force-rs-merkle")]
 fn rs_merkle_root<I>(leaves: I) -> H256
 where
 	I: Iterator<Item = Vec<u8>>,
 {
-	use rs_merkle::{algorithms::Sha256, Hasher, MerkleTree};
+	use rs_merkle::{Hasher, MerkleTree};
+
+	#[derive(Clone)]
+	pub struct Keccak256Algorithm {}
+
+	impl Hasher for Keccak256Algorithm {
+		type Hash = [u8; 32];
+
+		fn hash(data: &[u8]) -> [u8; 32] { sp_io::hashing::keccak_256(data).into() }
+	}
```

Comment:
Hopefully `KeccakHasher` is already implemented here: https://docs.rs/sp-core/latest/src/sp_core/hasher.rs.html#49-57
```suggestion
```

Reply:
needs to implement rs_merkle:Hasher